cli/tests/pcluster/config/test_update_policy.py

```diff
@@ -1701,46 +1701,74 @@ def test_condition_checker_managed_fsx(
 @pytest.mark.parametrize(
-    "base_config, target_config, change, expected_update_allowed, expected_fail_reason, expected_action_needed",
+    "key, path, old_value, new_value, expected_update_allowed, expected_fail_reason, expected_actions_needed",
     [
         pytest.param(
+            "Pools",
+            ["LoginNodes"],
             {
-                "LoginNodes": {"Pools": [{"Name": "mock-lp1"}]},
+                "Name": "pool-old",
+                "InstanceType": "t2.micro",
+                "GracetimePeriod": 3,
+                "Count": 1,
+                "Networking": {"SubnetIds": ["subnet-05cfbd48a49df385c"]},
+                "Ssh": {"KeyName": "xuanqi-us-east-2"},
             },
-            {
-                "LoginNodes": {"Pools": [{"Name": "mock-lp2"}]},
-            },
-            Change(
-                path=["LoginNodes"],
-                key="Pools",
-                old_value={"Name": "mock-lp1"},
-                new_value=None,
-                update_policy={},
-                is_list=True,
-            ),
+            None,
             False,
             "The update is not supported when login nodes are running",
             "Stop the login nodes by setting Count parameter to 0 "
             "and update the cluster with the pcluster update-cluster command",
             id="Login nodes must be stopped to remove a pool",
         ),
+        pytest.param(
+            "Pools",
+            ["LoginNodes"],
+            None,
+            {
+                "Name": "pool-new",
+                "InstanceType": "t2.micro",
+                "GracetimePeriod": 3,
+                "Count": 1,
+                "Networking": {"SubnetIds": ["subnet-05cfbd48a49df385c"]},
+                "Ssh": {"KeyName": "xuanqi-us-east-2"},
+            },
+            True,
+            "The update is not supported when login nodes are running",
+            "Stop the login nodes by setting Count parameter to 0 "
+            "and update the cluster with the pcluster update-cluster command",
+            id="Login pools can be added",
+        ),
     ],
 )
 def test_login_nodes_pools_policy(
     mocker,
-    base_config,
-    target_config,
-    change,
+    key,
+    path,
+    old_value,
+    new_value,
     expected_update_allowed,
     expected_fail_reason,
-    expected_action_needed,
+    expected_actions_needed,
 ):
-    update_policy = UpdatePolicy.LOGIN_NODES_POOLS
-    cluster = Cluster(name="mock-name", stack="mock-stack")
-    patch = ConfigPatch(cluster=cluster, base_config=base_config, target_config=target_config)
-    assert_that(update_policy.condition_checker(change, patch)).is_equal_to(expected_update_allowed)
-    assert_that(update_policy.fail_reason(change, patch)).is_equal_to(expected_fail_reason)
-    assert_that(update_policy.action_needed(change, patch)).is_equal_to(expected_action_needed)
+    cluster = dummy_cluster()
+    mocker.patch.object(cluster, "has_running_login_nodes", return_value=True)
```

Comment:
Can we make this a parametrize variable? Then we can cover these scenarios:
- Removal of a login node pool + `has_running_login_nodes=True`
- Removal of a login node pool + `has_running_login_nodes=False`
- Addition of a login node pool + `has_running_login_nodes=True`
- Addition of a login node pool + `has_running_login_nodes=False`

Reply:
Yes we can. I will address this comment later.